File: cai/v2/cai.c
#include "cai.h"
#include "random.h"
#include <stdio.h>

void cai_init(void){
    random_init();
}

int cai_generate_num(int min, int max){

    int num;

    max += 1;

    num = random_between(min, max);

    return num;
}

int cai_generate_string(int risposta, int risposta_esatta){

    int num = cai_generate_num(1, 4);
    int num_esatte = 0;
    int num_errate = 0;

    if (risposta == risposta_esatta){
        cai_risposta_esatta(num);
        num_esatte++;
    }

    else {
        cai_risposta_errata(num);
    }

    putchar('\n');

    return num_esatte;
}

void cai_risposta_esatta(int num){

    switch (num) {
        case 1:
            printf("Very good!");
            break;
        case 2:
            printf("Excellent!");
            break;
        case 3:
            printf("Nice work!");
            break;
        case 4:
            printf("Keep up the good work!");
            break;
        default:
            printf("Error");
            break;
    }
}

void cai_risposta_errata(int num){

    switch (num) {
        case 1:
            printf("No. Please try again.");
            break;
        case 2:
            printf("Wrong. Try once more.");
            break;
        case 3:
            printf("Don't give up!");
            break;
        case 4:
            printf("No. Keep trying.");
            break;
        default:
            printf("Error");
            break;
    }
}

File: cai/py/cai.py
import random

def generate_num(min, max):
    return random.randint(min, max)

def check_result_and_generate_string(risposta, risposta_esatta):

    num = generate_num(1, 4)
    esatta = 0

    if risposta == risposta_esatta:
        print_risposta_esatta(num)
        esatta = 1

    else:
        print_risposta_errata(num)

    print('\n')

    return esatta

def print_risposta_esatta(num):
    if num == 1:
        print("Very good!")
    elif num == 2:
        print("Excellent!")
    elif num == 3:
        print("Nice work!")
    elif num == 4:
        print("Keep up the good work!")
    else:
        print("Error")

def print_risposta_errata(num):
    if num == 1:
        print("No. Please try again.")
    elif num == 2:
        print("Wrong. Try once more.")
    elif num == 3:
        print("Don't give up!")
    elif num == 4:
        print("No. Keep trying.")
    else:
        print("Error")

if __name__ == '__main__':
  import doctest
  doctest.testmod()